src/inmanta/protocol/methods_v2.py

```diff
@@ -657,3 +657,57 @@ def resource_logs(
     :raise NotFound: This exception is raised when the referenced environment is not found
     :raise BadRequest: When the parameters used for filtering, sorting or paging are not valid
     """
+
+
+@typedmethod(path="/compilereport", operation="GET", arg_options=methods.ENV_OPTS, client_types=[ClientType.api], api_version=2)
+def get_compile_reports(
+    tid: uuid.UUID,
+    limit: Optional[int] = None,
+    first_id: Optional[uuid.UUID] = None,
+    last_id: Optional[uuid.UUID] = None,
+    start: Optional[datetime.datetime] = None,
+    end: Optional[datetime.datetime] = None,
+    filter: Optional[Dict[str, List[str]]] = None,
+    sort: str = "requested.desc",
+) -> List[model.CompileReport]:
+    """
+    Get the compile reports from an environment
+    :param tid: The id of the environment
+    :param limit: Limit the number of instances that are returned
+     :param first_id: The id to use as a continuation token for paging, in combination with the 'start' value,
+            because the order by column might contain non-unique values
+    :param last_id: The id to use as a continuation token for paging, in combination with the 'end' value,
+            because the order by column might contain non-unique values
+    :param start: The lower limit for the order by column (exclusive).
+                Only one of 'start' and 'end' should be specified at the same time.
+    :param end: The upper limit for the order by column (exclusive).
+                Only one of 'start' and 'end' should be specified at the same time.
+    :param filter: Filter the list of returned compile reports.
```

Comment:
I would add the overview of the filtering documentation somewhere centrally. 

It is OK to have it here, but if there is a general approach to filtering, that should be documented somewhere as well.